packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -75,16 +101,172 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
     return element.getChildren();
   }
 
-  public addCheckpointNode(
-    sourceFile: string,
-    typeRef: string,
-    line: number
-  ): CheckpointNode {
-    const cpNode = new CheckpointNode(sourceFile, typeRef, line);
+  // There's a limit to the number of checkpoints for a given user and
+  // that number is 5.
+  public canAddCheckpointNote(): Boolean {
+    return this.getChildren().length < MAX_ALLOWED_CHECKPOINTS;
+  }
+
+  public async addCheckpointNode(
```

Comment:
I think renaming this to getOrCreateCheckpointNode would be more succinct and let us remove the comment block on 117-119

Reply:
Done.